src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -160,6 +165,7 @@ public override void Start()
             connectionParameters.IsRelay = !this.nodeSettings.ConfigReader.GetOrDefault("blocksonly", false);
             connectionParameters.TemplateBehaviors.Add(new ChainHeadersBehavior(this.chain, this.chainState, this.loggerFactory));
             connectionParameters.TemplateBehaviors.Add(new AddressManagerBehavior(this.addressManager) { PeersToDiscover = 10 });
+            connectionParameters.TemplateBehaviors.Add(new PeerLifetimeBehavior(this.connectionManager, this.loggerFactory, this.peerLifetime));
```

Comment:
does the order of behaviors matter? if so, is CHB, AMB, PLB the right order?

Reply:
No order does not matter (and should not in my opinion)
In the case of CHA the logic is inside a filter so that gets called before all the other behaviours.